okhttp/src/test/java/okhttp3/ConnectionReuseTest.java

```diff
@@ -41,7 +42,11 @@
   @Rule public final OkHttpClientTestRule clientTestRule = new OkHttpClientTestRule();
 
   private HandshakeCertificates handshakeCertificates = localhost();
-  private OkHttpClient client = clientTestRule.client;
+  private OkHttpClient client;
+
+  @Before public void setUp() {
```

Comment:
This is really unfortunate, adding the setup methods.  I wonder whether we can avoid this boilerplate.  What is the effect of only replacing `.client` with `.newClient()`?  Does it negate pooling by eagerly creating a client for every test before running them?

Could we go the opposite way?  remove the field as a cache of the client and use the rule as the cache?  The cache knows both a) the test it is installed for, b) the lifecycle of the test.  And could have an optional param for second/third client instances?

Reply:
I'm more concerned if we start doing more of this and having setup in every test.  Currently the change looks like a big net positive

Reply:
Let's use the rule as a cache. That's a nice improvement.